SeaPkg/Tools/GenSeaArtifacts/gen_aux/src/auxgen.rs

```diff
@@ -409,13 +417,83 @@ impl AuxBuilder {
                 return Err(anyhow::anyhow!(e))
             }
         }
-        
+
+        for rule in &mut rules {
+            rule.resolve(&symbols, type_info)?;
+        }
+        Ok(rules)
+    }
+
+    /// Returns the symbols that are not excluded by the given configuration.
+    fn get_symbols(&self) -> anyhow::Result<Vec<Symbol>> {
+        let (symbols, filtered): (Vec<_>, Vec<_>) = self
+            .symbols
+            .iter()
+            .cloned()
+            .partition(|symbol| !self.excluded_symbols.contains(&symbol.name));
+
+        for symbol in filtered {
+            println!("Symbol: {:?} Skipped... Excluded via `excluded_symbols` in config file", symbol);
+        }
+
+        Ok(symbols)
+    }
+
+    /// Resolves and returns the key symbols that are not excluded by the given configuration.
+    fn get_key_symbols(&self) -> anyhow::Result<Vec<KeySymbol>> {
+        let symbols = self.get_symbols()?;
+        let mut key_symbols = self.key_symbols.clone();
+
+        for symbol in &mut key_symbols {
+            symbol.resolve(&symbols)?;
+        }
+        Ok(key_symbols)
+    }
+
+    /// Auto generates additional Content rules for padding that is present in the image.
+    fn auto_generate_entry_headers(&self, report: CoverageReport) -> anyhow::Result<Vec<ImageValidationEntryHeader>> {
+        let mut entry_headers = Vec::new();
+
+        let rules: Vec<ValidationRule> = report.segments()
+            .iter()
+            .filter(|segment| !segment.covered() && segment.symbol().is_empty())
+            .map(|segment| {
+                let mut rule = ValidationRule::new(String::from(""));
+                let size = segment.end() - segment.start();
+                rule.validation = ValidationType::Content { content: vec![0; size as usize] };
+
+                let name = if segment.reason() == "Padding" {
+                    String::from("Section Padding")
+                } else {
+                    String::from("Symbol Padding")
+                };
+
+                rule.symbol_info = Some(Symbol {
+                    name,
+                    address: segment.start() as u32,
+                    type_info: TypeInfo::one(size as u64, None),
+                    symbol_type: SymbolType::None,

```

Comment:
`section_characteristics: SectionCharacteristics::default()` needs to be added to be compatible with the latest changes.

Reply:
Added as suggested. Thank you!